event.go

```diff
@@ -95,6 +95,19 @@ func WithMetadata(metadata map[string]interface{}) EventOption {
 	}
 }
 
+// ForCommand adds relevant command data when crating an event.
+// Currently it adds the command type and optionally a command ID (if the
+// CommandIDer interface is implemented).
+func ForCommand(cmd Command) EventOption {
```

Comment:
@hansbogert I'm not perfectly happy with this, but it's a start. The problem is that the `AppendEvent()` method in the aggregate base doesn't know anything about the command as it is now. This was the least intrusive and most flexible solution I could think of, but it does require using the `ForCommand()` option when creating events in all places in aggregates where `AppendEvent()` is used.

Reply:
Not sure if I understand your own critique on this; as a control question, the same goes for `NewEvent(...)` right?

Reply:
Yes, it does. But the aggregate base calls new event for you, thus we need to manually add `ForCommand()` to all instances of `AppendEvent()`. If it's missed there will be no correlation with the command stored in the metadata. Thats the part I'm not happy with, the possibility of forgetting to add `ForCommand()` and breaking parts that rely on command metadata in events. An option would be to always pass the originating command to `AppendEvent()`, which would guarantee that we have command metadata at all times.

Reply:
>  An option would be to always pass the originating command to AppendEvent(), which would guarantee that we have command metadata at all times.

But the downside is that you want to have it optional?

Reply:
Yes, if possible it would be good to keep the API backward compatible.

Reply:
I think it's okay for now. Maybe we'll find a more elegant way.